initial draft for OSISM 7.0.3 minor release (#924)
SovereignCloudStack/website: _i18n/en/_posts/blog/2024-05-02-incremental-efforts-osism-703.md

  [OSISM issue #921](https://github.com/osism/issues/issues/921). Deleting the health-monitor before the
  members or using `openstack loadbalancer delete --cascade` avoids this issue.
* With amphora loadbalancers, we can end up in situations that LB deletion does no longer work due to
  a failover or a failed creation of the vrrp port. This is tracked in
  [OSISM issue #925](https://github.com/osism/issues/issues/925). An upstream fix exists and a backport
  is already underway.

Quickly after the initial release OSISM released [7.0.1](https://release.osism.tech/notes/7.html#id2) that
contained a bugfixes for two of these issues:

* Backport of [https://review.opendev.org/c/openstack/octavia/+/896995](https://review.opendev.org/c/openstack/octavia/+/896995)
to fix errors when deleting LB with broken amphorae.

Bugfix for [OSISM#890](https://github.com/osism/issues/issues/890) (octavia (ovn) does not find existing subnet)
by enabling the use of the custom CA for octavia user session queries with the following PR: https://github.com/osism/container-images-kolla/pull/412

The third issue is part of the OSISM 7.0.3 release.

## OSISM 7.0.3

> OSISM 7.0.3 is released and brings a fix for the leaking of octavia ports that was reported in [OSISM#921](https://github.com/osism/issues/issues/921).

Reviewer: Unfortunately, the upstream backports didn't make it... :(

But we fixed https://bugs.launchpad.net/octavia/+bug/2057751.

Alongside this important bugfix the next minor release of OSISM brings the following noteworthy items to the table:

> * ....

Reviewer: I think we should add the issue with the service restart delay. We have a playbook to workaround the issue before an upgrade.

The other nice thing we added are the improvements in the Ceph plays that will improve things when adding new nodes + new OSDs.

  members or using `openstack loadbalancer delete --cascade` avoids this issue.
* With amphora loadbalancers, we can end up in situations that LB deletion does no longer work due to
  a failover or a failed creation of the vrrp port. This is tracked in
  [OSISM issue #925](https://github.com/osism/issues/issues/925). An upstream fix exists and a backport
  is already underway.

Quickly after the initial release OSISM released [7.0.1](https://release.osism.tech/notes/7.html#id2) that
contained a bugfixes for two of these issues:

* Backport of [https://review.opendev.org/c/openstack/octavia/+/896995](https://review.opendev.org/c/openstack/octavia/+/896995) to fix errors when deleting LB with broken amphorae.

* Bugfix for [OSISM#890](https://github.com/osism/issues/issues/890) (octavia (ovn) does not find existing subnet) by enabling the use of the custom CA for octavia user session queries with the following PR: https://github.com/osism/container-images-kolla/pull/412

A plethora of fixes and additions were done since the 7.0.2 release a few weeks ago - as such today 7.0.3 was released by the OSISM team.

## OSISM 7.0.3

> While OSISM 7.0.3 does not yet bring a fix for the leaking of octavia ports that was reported in [OSISM#921](https://github.com/osism/issues/issues/921) since this is pending some upstream work, it does bring a fix for a [bug when creating a fully-populated LB with allowed_cidr](https://bugs.launchpad.net/octavia/+bug/2057751).

Reviewer 2: OK, this surprises me. I will test Fernandos patch again to confirm whether or not it addresses the issue. Or are you saying that the patch did not make it into the rebuilt images?

Reviewer: Backports to stable branches still pending. It's only fixed @ master.

Alongside this bugfix the next minor release of OSISM brings the following noteworthy items to the table:

* During the preparation of the upgrades of the regions of the PCO a bug ([osism/issues#937](https://github.com/osism/issues/issues/973)) has been noticed which leads to a delay of up to 2 minutes between the necessary container stops and starts. This is due to a bug in the service units of all Kolla services. The bug is fixed in the current release. To avoid the delay during an upgrade, a fix must be applied in advance for all srvice units from Kolla: `$ osism apply fix-gh937`.
* The `ceph_cluster_fsid` parameter is now generated automatically. It can be removed from `environments/configuration.yml`. The automatically generated `ceph_clusterfs_fsid` parameter is set to the value of the `fsid` parameter from `environments/ceph/configuration.yml`.
* Versions not yet pinned in the manager environment of the configuration repository (Ansible collections, `osism/cfg-generics`, ..) are now automatically pinned during synchronisation with gilt overlay. This also applies to the osism update manager script.

> * The Docker version and the Docker CLI version can now also be managed via `osism/cfg-generics`. It is recommended to pin the Docker version in `environments/configuration.yml`.

Reviewer 2: Why? To avoid unnecessary and uncoordinated restarts of containers when docker gets updated?

Reviewer: We try to pin all version that have an impact on the service quality. This has not been the case for Docker so far.

* Monitoring services are now activated by default for the internal Kubernetes cluster.

> * Kubernetes Cluster API for the 1.30 series is available. They are now provided directly with `osism manage image clusterapi`. This means that Kubernetes Cluster API images are now available for series 1.27, 1.28, 1.29 and 1.30.

Reviewer 2: ```suggestion
* Kubernetes Cluster API images for the Kubernetes (K8s) 1.30 series is available. They are now provided directly with `osism manage image clusterapi`. This means that K8s Cluster API images are now available for series 1.27, 1.28, 1.29 and 1.30.
```

Author: thanks.

> * All Ansible collections have been prepared for use with Ubuntu 24.04. It is currently not recommended to upgrade existing clusters to Ubuntu 24.04 or to install new clusters with Ubuntu 24.04. There will be a note in the release notes from which Ubuntu 24.04 can be used.

Reviewer 2: I don't get this statement. What is the relationship between (Kubernetes?) clusters and ansible collections. SCS k8s clusters don't use ansible, but shell scripts or better cluster-stack-operator and CAPI+CAPO to manage clusters. We could create Ubu24.04 CAPI images or a Ubu24.04 management host, but I still don't see any relationship with ansible collections. What clusters?

Apart from my ignorance, the last sentence needs improvement.
Maybe: `There will be a note in future release notes that announces the full support of Ubuntu 24.04.`
Maybe it also helps then to explicitly mention the status of Ubuntu 24.04 currently being a technical preview?

Reviewer: This is a separate point in a list of changes. All our Ansible collections have been prepare for use with Ubuntu 24.04. This is not related to the CAPI images.

Author: removed the sentence about the ansible collections.

> This list is by far not complete, a complete list is part of the release notes are available on [releases.osism.tech](https://release.osism.tech/notes/7.html).

Reviewer 2: ```suggestion
This list only mentions the highlights; a complete list is part of the release notes are available on [releases.osism.tech](https://release.osism.tech/notes/7.html).
```
(The value of an incomplete list may be debated. A highlight selection definitely has value.)

Author: thanks.

> Important note: The OpenStack service images for Octavia and Nova have been rebuilt. Upgrades of the Octavia and Nova services are recommended. No upgrades of other OpenStack and associated infrastructure services such as MariaDB or RabbitMQ are required.

Reviewer 2: It would be valuable if we can mention that the update to OSISM 7.0.3 will thus cause short downtimes (15s or so) for the nova and octavia APIs, but not for any other service and no disruption for running workloads.
(@berendt should better confirm this, this is only my guess from a bit of CiaB operations experience ...)

Reviewer: I'll add this as a general note in the Upgrade Guide.
